cmd/bb_storage/main.go

```diff
@@ -193,7 +197,7 @@ func main() {
 					}
 				}))
 	}()
-
+	readyGroup.Wait()
 	lifecycleState.MarkReadyAndWait()
```

Comment:
Could we do this as part of MarkReadyAndWait()?

I think that the main annoyance right now is that we have one instance of `health.Server` per RPC server, which is why you're trying to solve this as part of NewServersFromConfigurationAndServe(). But what if we eliminated that? I think it's completely safe to call `health.NewServer()` exactly once as part of `global.ApplyConfiguration()`, and somehow get that passed along as we create gRPC servers.

What are your thoughts?

Reply:
It would be great, but at the moment `grpc.proto` defines individual service names for each server. I don't know who uses such configuration. Would you like to remove it or do you have other ideas?
```protobuf
  // Service name for health check requests. The gRPC server will
  // report itself healthy for this service via the grpc.health.v1
  // protocol.
  string health_check_service = 7;
```

Reply:
Alternatively, `global.ApplyConfiguration` can provide a factory function which is passed to `NewServersFromConfigurationAndServe`. Then everything will still work.

Reply:
Indeed! That sounds like a good solution. We already provide a factory for clients as well.

Reply:
I now realised that the `readyGroup.Wait()` call is there to make sure that `NewServersFromConfigurationAndServe` has finished with all the initialisation, e.g. started listening to all the ports.

The error returned from `NewServersFromConfigurationAndServe` serves dual purposes at the moment, both startup errors and errors during runtime (from `s.Serve()`). It is wrapped with "xyz gRPC server failure: " in `main.go` and I don't want to duplicate that error wrapping. Any ideas?